### PR TITLE
Day2

### PR DESCRIPTION
**Pull request by rex-9**
- Added a column `species` of type `string` to `pets` table. Modified `schema.sql` file.
- Inserted the following data:
  - Pet: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Pet: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Pet: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Pet: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Pet: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Pet: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Pet: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Added these insertions to the `data.sql` file.

----

- Inside a transaction updated the `pets` table by setting the `species` column to `unspecified`. Verified that change was made. Then rolled back the change and verify that the `species` columns went back to the state before the transaction.
- Inside a transaction:
  - Updated the `pets` table by setting the `species` column to `Digimon` for all pets that have a name ending in `mon`.
  - Updated the `pets` table by setting the `species` column to `Pokemon` for all pets that don't have `species` already set.
  - Committed the transaction.
  - Verified that change was made and persists after commit.
- **The screenshot is  `species.png`.
species
**
- Now, took a deep breath and... **Inside a transaction** deleted all records in the `pets` table, then rolled back the transaction.
- After the rollback verify if all records in the `pets` table still exist. After that, I started breathing as usual ;) 
- **The screenshot is `delete.png`.**
delete

- Inside a transaction:
  - Deleted all pets born after Jan 1st, 2022.
  - Created a savepoint for the transaction.
  - Updated all pets' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Updated all pets' weights that are negative to be their weight multiplied by -1.
  - Committed transaction
- **The screenshot is `weights.png`.**
weights

- Added all these transactions to the `queries.sql` file.

----
- Wrote queries to answer the following questions:
  - How many pets are there?
  - How many pets have never tried to escape?
  - What is the average weight of pets?
  - Who escapes the most, neutered or not neutered pets?
  - What is the minimum and maximum weight of each type of pet?
  - What is the average number of escape attempts per pet type of those born between 1990 and 2000?
- **The screenshot is `aggregates.png`.**
aggregates

- Added these queries to the `queries.sql` file.
- Added screenshots with the results of queries and with the results of your play with transactions to the Pull Request description.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+INSERT INTO pets (name, birth_date, escape_attempts, neutered, weight_kg) VALUES ('Agumon', DATE '2020-02-03', 0, true, 10.23);
+INSERT INTO pets (name, birth_date, escape_attempts, neutered, weight_kg) VALUES ('Gabumon', DATE '2018-11-15', 2, true, 8);
+INSERT INTO pets VALUES (3, 'Pikachu', DATE '2021-01-07', 1, false, 15.04);
+INSERT INTO pets VALUES (4, 'Devimon', DATE '2017-05-12', 5, true, 11);
+
+-- day 2
+INSERT INTO pets (id, name, birth_date, escape_attempts, neutered, weight_kg) VALUES (5, 'Charmander', DATE '2020-02-08', 0, false, 11);
+INSERT INTO pets (id, name, birth_date, escape_attempts, neutered, weight_kg) VALUES (6, 'Plantmon', DATE '2021-11-15', 2, true, 5.7);
+INSERT INTO pets (id, name, birth_date, escape_attempts, neutered, weight_kg) VALUES (7, 'Squirtle', DATE '1993-04-02', 3, false, 12.13);
+INSERT INTO pets (id, name, birth_date, escape_attempts, neutered, weight_kg) VALUES (8, 'Angemon', DATE '2005-06-12', 1, true, 45);
```
Comment (cynthiainga) [inline data.sql]: - [ ] Please kindly add the `-` to your `weight` as asked on the project requirements the values are `-11.0, -5.7, -12.13, -45.0` 🙃 🙃

File: queries.sql [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+SELECT name, escape_attempts from pets WHERE weight_kg > 10.5;
+SELECT * from pets WHERE neutered is true;
+SELECT * from pets WHERE name != 'Gabumon';
+SELECT * from pets WHERE weight_kg between 10.4 and 17.3;
+
+-- day 2
+BEGIN;
+ALTER TABLE pets
+RENAME COLUMN species TO unspecified; -- rename column from species to unspecified
+ROLLBACK; -- screenshot is species.png
```
Comment (cynthiainga) [inline queries.sql:15]: - [ ] After `ROLLBACK` you should select species from your pets to verify if the change was undone 😉

File: queries.sql [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+UPDATE pets
+SET species = 'Digimon'
+WHERE name LIKE '%mon'; -- Set species of Digimon to the pets that have 'mon' in the end of their name
+
+UPDATE pets
+SET species = 'Pokemon'
+WHERE species ISNULL; -- Set species of Pokemon to the pets WHERE species IS NULL;
+
```
Comment (cynthiainga) [inline queries.sql:26]: - [ ] After updating your table you should select species from pets to verify if that change was made 🙃

File: queries.sql [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+COMMIT;
+
```
Comment (cynthiainga) [inline queries.sql]: - [ ] After committing you should select species from pets to verify if the change persists after commit 😉 😉

File: queries.sql [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+BEGIN;
+DELETE FROM pets; -- Delete all rows from the pets table
```
Comment (cynthiainga) [inline queries.sql:30]: - [ ] After deleting you should select all from pets to verify if that change was made 🙃

File: queries.sql [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ROLLBACK; -- screenshot is delete.png
+
```
Comment (cynthiainga) [inline queries.sql]: After the rollback, you should select all from pets to verify if that change was made 🙃

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @rex-9, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Descriptive PR ✔️ 
- [ ] Good commit history ✔️ 

## Required Changes ♻️

- [ ] Please kindly add screenshots of your queries results to the pull request description as asked in the project requirements 🙃 😉
- [ ] Please kindly also check the comments under the review 👇 👇

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @rex-9 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Great job implementing all the changes by the previous reviewer ✅
- Professional README file ✅
- Using GitHub flow ✅

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**